examples/shader/compute_mesh.rs

```diff
@@ -0,0 +1,297 @@
+//! This example shows how to initialize an empty mesh with a Handle
+//! and a render-world only usage. That buffer is then filled by a
+//! compute shader on the GPU without transferring data back
+//! to the CPU.
+//!
+//! The `mesh_allocator` is used to get references to the relevant slabs
+//! that contain the mesh data we're interested in.
+//!
+//! This example does not remove the `GenerateMesh` component after
+//! generating the mesh.
+
+use bevy::{
+    asset::RenderAssetUsages,
+    color::palettes::tailwind::RED_400,
+    mesh::Indices,
+    prelude::*,
+    render::{
+        extract_component::{ExtractComponent, ExtractComponentPlugin},
+        mesh::allocator::MeshAllocator,
+        render_graph::{self, RenderGraph, RenderLabel},
+        render_resource::{
+            binding_types::{storage_buffer, uniform_buffer},
+            *,
+        },
+        renderer::{RenderContext, RenderQueue},
+        Render, RenderApp, RenderStartup,
+    },
+};
+
+/// This example uses a shader source file from the assets subdirectory
+const SHADER_ASSET_PATH: &str = "shaders/compute_mesh.wgsl";
+
+fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins,
+            ComputeShaderMeshGeneratorPlugin,
+            ExtractComponentPlugin::<GenerateMesh>::default(),
+        ))
+        .insert_resource(ClearColor(Color::BLACK))
+        .add_systems(Startup, setup)
+        .run();
+}
+
+// We need a plugin to organize all the systems and render node required for this example
+struct ComputeShaderMeshGeneratorPlugin;
+impl Plugin for ComputeShaderMeshGeneratorPlugin {
+    fn build(&self, app: &mut App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app
+            .init_resource::<Chunks>()
+            .add_systems(
+                RenderStartup,
+                (init_compute_pipeline, add_compute_render_graph_node),
+            )
+            .add_systems(Render, prepare_chunks);
+    }
+    fn finish(&self, app: &mut App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+        render_app
+            .world_mut()
+            .resource_mut::<MeshAllocator>()
+            .extra_buffer_usages = BufferUsages::STORAGE;
+    }
+}
+
+/// Holds a handle to the empty mesh that should be filled
+/// by the compute shader.
+#[derive(Component, ExtractComponent, Clone)]
+struct GenerateMesh(Handle<Mesh>);
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // a truly empty mesh will error if used in Mesh3d
```

Comment:
We should highlight how it's _very_ important that you allocate the maximum amount of space that you need, and not less.

If you don't need to fill the whole space, you can write NaN triangles to the rest of the slice.